src/main/java/org/generationcp/middleware/service/impl/study/StudyInstanceServiceImpl.java

```diff
@@ -337,78 +334,72 @@ public InstanceDescriptorData addInstanceDescriptorData(final InstanceDescriptor
-				environmentVariables.addAll(environmentDetails);
-				environmentParameters.addAll(this.createGeolocationVariables(environmentVariables, instanceId));
-				studyDetailsDto.setEnvironmentParameters(environmentParameters);
+			final List<MeasurementVariable> environmentVariables = new ArrayList<>(environmentConditions);
+			environmentVariables.addAll(environmentDetails);
+			environmentParameters.addAll(this.createGeolocationVariables(environmentVariables, instanceId));
+			studyDetailsDto.setEnvironmentParameters(environmentParameters);
 
-				final Map<String, String> properties = new HashMap<>();
-				variableIds = environmentVariables.stream().map(MeasurementVariable::getTermId)
-					.collect(Collectors.toList());
-				properties.put("studyObjective", studyMetadata.getStudyObjective() == null ? "" : studyMetadata.getStudyObjective());
-				final Map<String, String> geolocationMap = this.daoFactory.getGeolocationPropertyDao()
-					.getGeolocationPropsAndValuesByGeolocation(Collections.singletonList(instanceId), variableIds).get(instanceId);
-				if (geolocationMap != null) {
-					properties.putAll(geolocationMap);
-				}
+			final Map<String, String> properties = new HashMap<>();
+			variableIds = environmentVariables.stream().map(MeasurementVariable::getTermId)
+				.collect(Collectors.toList());
+			properties.put("studyObjective", studyMetadata.getStudyObjective() == null ? "" : studyMetadata.getStudyObjective());
+			final Map<String, String> geolocationMap = this.daoFactory.getGeolocationPropertyDao()
+				.getGeolocationPropsAndValuesByGeolocation(Collections.singletonList(instanceId), variableIds).get(instanceId);
+			if (geolocationMap != null) {
+				properties.putAll(geolocationMap);
+			}
 
-				final Map<Integer, Map<String, String>> projectPropMap =
-					this.daoFactory.getProjectPropertyDAO().getProjectPropsAndValuesByStudyIds(
-						Collections.singletonList(studyMetadata.getNurseryOrTrialId()));
-				if (projectPropMap.containsKey(studyMetadata.getNurseryOrTrialId())) {
-					properties.putAll(projectPropMap.get(studyMetadata.getNurseryOrTrialId()));
-				}
-				studyDetailsDto.setAdditionalInfo(properties);
-				return studyDetailsDto;
+			final Map<Integer, Map<String, String>> projectPropMap =
+				this.daoFactory.getProjectPropertyDAO().getProjectPropsAndValuesByStudyIds(
+					Collections.singletonList(studyMetadata.getNurseryOrTrialId()));
+			if (projectPropMap.containsKey(studyMetadata.getNurseryOrTrialId())) {
+				properties.putAll(projectPropMap.get(studyMetadata.getNurseryOrTrialId()));
 			}
-			return null;
-		} catch (final MiddlewareQueryException e) {
-			final String message = "Error with getStudyDetailsForGeolocation() query with instanceId: " + instanceId;
-			StudyInstanceServiceImpl.LOG.error(message, e);
-			throw new MiddlewareQueryException(message, e);
+			studyDetailsDto.setAdditionalInfo(properties);
+			return studyDetailsDto;
 		}
+		return null;
```

Comment:
Returning null should be used very sparingly. Can we try to convert this API to:
Optional<StudyDetailsDto> getStudyDetailsByInstance(final Integer instanceId)